### ISSUE TITLE
Export triggers window.onbeforeunload event

### ISSUE THREAD

**Issue by TorsteinHonsi** [highcharts/highcharts#2981]
The solution would be to run the export in a hidden iframe. 

Drop-in fix: http://jsfiddle.net/highcharts/6GAw6/

``` javascript
// Drop-in fix for running the export in an iframe
Highcharts.wrap(Highcharts, 'post', function (proceed, url, data, formAttributes) {
    if (!Highcharts.$exportFrame) {
        Highcharts.$exportFrame = 
            $('<iframe id="hc-export-frame" style="display:none">').appendTo(document.body);   
    }
    formAttributes = Highcharts.merge({ target: 'hc-export-frame' }, formAttributes);
    proceed.call(this, url, data, formAttributes);
});
```

**Comment by rolandas-valantinas** [comment]
Thanks, this was the only solution to solve that stealthy trigger of `onbeforeunload` by exporting module.
It's such a joke it's not been fixed till now!

**Comment by TorsteinHonsi** [comment]
> It's such a joke it's not been fixed till now!

Well... Fixing an issue that hasn't received a single comment in three and a half year isn't top priority.

I rewrote the fix to non-jQuery code. But there's still a problem with it, in Chrome it opens up a new tab briefly before opening the image.

**Comment by Denyllon** [comment]
It occurs because the form submitting in _H.post()_ function. Every submit() of form triggers the unbeforeunload event, so I did a few prototypes overrides in which I changed the method of sending the data to server.
I used common XMLHttpRequest to generate the file on export server, and downloading it when it's already generated.
Additionally, I defined the **beforeExporting** and **afterExporting** events, which can be used from chart configuration object level.

 JSFiddle example: [http://jsfiddle.net/daniel_s/7qffmhnu/](http://jsfiddle.net/daniel_s/7qffmhnu/)

Code below:
```javascript
(function(H) {
	
  // Workaround to download the exported chart. It creates a element with reference to downloading the generated chart. Url passed as an argument
    var download = function(url) {
      var a = document.createElement("a")
      a.style = 'display: none'
      a.href = url
      a.setAttribute('download', '')
      a.innerText = "Link"
      document.body.appendChild(a)
      a.click()
      a.remove()
    };

  // Fix to #7653
  H.SVGRenderer.prototype.getStyle = function(style) {
    this.style = H.extend({

      fontFamily: 'Lucida Grande, Lucida Sans Unicode, ' +
        'Arial, Helvetica, sans-serif',
      fontSize: '12px'

    }, style);
    return this.style;
  }

  H.post = function(url, data, formAttributes, chart) {
    // Trigger beforeExporting event
    H.fireEvent(chart, 'beforeExporting')

    // Build formData
    var dataString = encodeURI('async=true&type=' + data.type + '&svg=' + data.svg + '&width=' + data.width + '&scale=' + data.scale)

    // Choose type of request
    function createCORSRequest(method, url) {
      var xhr = new XMLHttpRequest();
      if ("withCredentials" in xhr) {
        xhr.open(method, url);
      } else if (typeof XDomainRequest != "undefined") {
        xhr = new XDomainRequest();
        xhr.open(method, url);
      } else {
        xhr = null;
      }
      return xhr
    }

    // Send AJAX
    var xhr = createCORSRequest('POST', url);
    if (!xhr) {
      throw new Error('CORS not supported');
    }
    xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
    xhr.onload = function() {
      download(url + xhr.response)
        //Trigger afterExporting event
      H.fireEvent(chart, 'afterExporting')
    }
    xhr.send(dataString)
  }

  // Call post() function with Chart object passed as the third argument to make afterExporting event works
  H.Chart.prototype.exportChart = function(exportingOptions, chartOptions) {

    var svg = this.getSVGForExport(exportingOptions, chartOptions);

    // merge the options
    exportingOptions = H.merge(this.options.exporting, exportingOptions);

    // do the post
    H.post(exportingOptions.url, {
      filename: exportingOptions.filename || 'chart',
      type: exportingOptions.type,
      width: exportingOptions.width || 0, // IE8 fails to post undefined correctly, so use 0
      scale: exportingOptions.scale,
      svg: svg
    }, exportingOptions.formAttributes, this); // Add chart as third argument

  }

})(Highcharts)
```
